### PR TITLE
Add tutorial for transition from legacy SPA.

### PR DESCRIPTION
**Pull request by jgosmann**
**Motivation and context:**
This adds a tutorial describing what changed in nengo_spa compared to the legacy SPA implementation. My presentation on Friday will be based on this.

There are a few spots where links are missing because the documentation that should be linked to has not been written (or updated) yet. I think this PR should be merged anyways because adding all the remaining documentation is a larger job that will be done in the (hopefully near) future.

**Interactions with other PRs:**
none

**How has this been tested?**
Ran all cells in the notebook manually. There is also a test that checks that no exceptions are raised in the notebook. I will add a commit to add the notebook to the list of slow tests once the CI builds have completed to not waste time on this for each commit, but at least we run it once in the full test matrix this way.

**How long should this take to review?**

- Average (neither quick nor lengthy)

**Types of changes:**

- Bug fix (non-breaking change which fixes an issue)
- Non-code change (touches things like tests, documentation, build scripts)

**Checklist:**

- [x] I have read the **CONTRIBUTING.rst** document.
- [x] I have updated the documentation accordingly.
- [ ] I have included a changelog entry.
- [n/a] I have added tests to cover my changes.
- [x] All new and existing tests passed.

**Still to do:**
- [x] Add the notebook to the list of slow examples.

### CODE REVIEW

**Comment by jgosmann** [comment]
Updated the PR description, this is ready for review now.

**Comment by jgosmann** [comment]
Note that there are some somewhat unrelated commits fixing tests (and I'm too lazy to do separate PRs for those).

File: nengo_spa/examples/intro_coming_from_legacy_spa.ipynb [hunk 1]
```diff
@@ -0,0 +1,1183 @@
...
+    "collapsed": true
+   },
+   "source": [
+    "## New associative memory classes\n",
+    "\n",
+    "To reduce the number of arguments of the associative memory class it has been split into two classes. `spa.ThresholdingAssocMem` is the applying a simple threshold, `spa.WTAAssocMem` performs a winner-take-all clean-up with lateral inhibitory connections. There is also a new type of associative memory `spa.IAAssocMem` based on independent accumulators, that also exhibits winner-take-all behaviour, but with different dynamics. (TODO link CogSci paper once published)\n",
+    "\n",
+    "To implement auto-associative memories, the `input_keys` and `output_keys` arguments have been replaced by a single `mapping` arguments.\n",
+    "\n",
+    "TODO link associative memory documentation/tutorial."
```
Comment (Seanny123) [inline nengo_spa/examples/intro_coming_from_legacy_spa.ipynb]: Is this tutorial already written in `nengo_spa/nengo_spa/examples/associative_memory.ipynb`? How do we usually link between examples?

Comment (jgosmann) [inline nengo_spa/examples/intro_coming_from_legacy_spa.ipynb]: It needs updating. Not sure about how to do the linking. Maybe the link can go to the online documentation that should include the examples like in core Nengo.

Comment (Seanny123) [inline nengo_spa/examples/intro_coming_from_legacy_spa.ipynb]: I'm fine with figuring this out later.

**Pull request merged**